### PR TITLE
Tta #76 server prisma routes controllers event emitters

### PR DESCRIPTION
**Pull request by mnmin**

### CODE REVIEW

File: prisma/seed.js [hunk 1]
```diff
@@ -235,6 +235,172 @@ async function seed() {
...
+  const curriculums = []
+  const createCurriculum = async () => {
+    const newCurriculum = await prisma.curriculum.create({
+      data: {
+        name: 'Javascript',
+        description:
+          'Learn the JavaScript fundamentals you will need for front-end or back-end development'
+      }
+    })
+    curriculums.push(newCurriculum)
```
Comment (Auenc) [inline prisma/seed.js:273]: Don't push the created curriculum to an array, instead return it from the function, and have something else add it to an array. 

Adding to an array from inside a function is a side-effect, and it is always best to try and reduce them as much as possible

File: prisma/seed.js [hunk 2]
```diff
@@ -235,6 +235,172 @@ async function seed() {
...
+          `Objective 5 for module-${iterator} Curriculum-${curriculum.id}`
+        ],
+        curriculums: {
+          connect: {
+            id: curriculum.id
+          }
+        }
+      }
+    })
+    modules.push(newModule)
```
Comment (Auenc) [inline prisma/seed.js:296]: same as above

File: prisma/seed.js [hunk 3]
```diff
@@ -235,6 +235,172 @@ async function seed() {
...
+          `Objective 1 for Unit-${iterator} Module-${module.id} `,
+          `Objective 2 for Unit-${iterator} Module-${module.id} `,
+          `Objective 3 for Unit-${iterator} Module-${module.id} `,
+          `Objective 4 for Unit-${iterator} Module-${module.id} `,
+          `Objective 5 for Unit-${iterator} Module-${module.id}`
+        ],
+        moduleId: module.id
+      }
+    })
+    units.push(newUnit)
```
Comment (Auenc) [inline prisma/seed.js:315]: same as above

File: prisma/seed.js [hunk 4]
```diff
@@ -235,6 +235,172 @@ async function seed() {
...
+          }
+        },
+        lessons: {
+          connect: {
+            id: lesson.id
+          }
+        }
+      }
+    })
+    lessonPlans.push(newLessonPlan)
```
Comment (Auenc) [inline prisma/seed.js:393]: same as above

File: src/eventEmitter/index.js [hunk 5]
```diff
@@ -55,4 +70,53 @@ myEmitter.on('delete-exercise', (exercise, user) =>
   createDeleteExerciseEvent(exercise, user)
 )
 
+// CURRICULUM
+myEmitter.on('create-curriculum', (curriculum, user) =>
```
Comment (Auenc) [inline src/eventEmitter/index.js:74]: Split this file up so functions/events relating to curriculums, modules, units, etc are all in their own files.

File: src/eventEmitter/utils.js [hunk 6]
```diff
@@ -230,6 +230,290 @@ export const createDeleteExerciseEvent = async (exercise, user) => {
   })
 }
 
+export const createCurriculumCreatedEvent = async (curriculum, user) => {
```
Comment (Auenc) [inline src/eventEmitter/utils.js:233]: Same as above, have functions for the different models in their own files

**Review by Auenc — DISMISSED** [review]

File: prisma/seed.js (hunk 1 above)
Comment (mnmin) [reply prisma/seed.js]: Updated

File: prisma/seed.js (hunk 2 above)
Comment (mnmin) [reply prisma/seed.js]: Updated

File: prisma/seed.js (hunk 3 above)
Comment (mnmin) [reply prisma/seed.js]: Updated

File: prisma/seed.js [hunk 7]
```diff
@@ -235,6 +235,172 @@ async function seed() {
...
+          `Objective 1 for lesson-${iterator} Unit-${unit.id}`,
+          `Objective 2 for lesson-${iterator} Unit-${unit.id} `,
+          `Objective 3 for lesson-${iterator} Unit-${unit.id} `,
+          `Objective 4 for lesson-${iterator} Unit-${unit.id} `,
+          `Objective 5 for lesson-${iterator} Unit-${unit.id}`
+        ],
+        unitId: unit.id
+      }
+    })
+    lessons.push(newLesson)
```
[the remark this answers is not in the document]
Comment (mnmin) [reply prisma/seed.js]: Updated

File: prisma/seed.js [hunk 8]
```diff
@@ -235,6 +235,172 @@ async function seed() {
...
+          `Objective 5 for exercise ${iterator} lesson-${lesson.id}`
+        ],
+        lessons: {
+          connect: {
+            id: lesson.id
+          }
+        }
+      }
+    })
+    exercises.push(newExercise)
```
[the remark this answers is not in the document]
Comment (mnmin) [reply prisma/seed.js]: Updated

File: prisma/seed.js (hunk 4 above)
Comment (mnmin) [reply prisma/seed.js]: Updated

**Review by dearshrewdwit — APPROVED** [review]
🤞

**Review by ning905 — APPROVED** [review]

**Pull request merged**